hooks/hookexecution/outcome.go

```diff
@@ -0,0 +1,77 @@
+package hookexecution
+
+import (
+	"time"
+
+	"github.com/prebid/prebid-server/hooks/hookanalytics"
+	"github.com/prebid/prebid-server/hooks/hookstage"
+)
+
+type Status string
+
+const (
+	StatusSuccess          Status = "success"
+	StatusTimeout          Status = "timeout"
+	StatusFailure          Status = "failure"           // expected module-side failure occurred during hook execution
+	StatusExecutionFailure Status = "execution_failure" // unexpected failure occurred during hook execution
+)
+
+type Action string
+
+const (
+	ActionUpdate   Action = "update"
+	ActionReject   Action = "reject"
+	ActionNoAction Action = "no_action"
```

Comment:
We were just matching PBS-java. I did not quite understand your question about exposing externally, but the `no_action` will be used when a hook does not change the payload (there are no mutations to be applied) because f.i. it just validates some data and this action result will be stored in the debug info